test/webgpu_required_limits.c

```diff
@@ -0,0 +1,97 @@
+#include <assert.h>
+#include <emscripten.h>
+#include <stdio.h>
+#include <webgpu/webgpu.h>
+
+int adapter_limit_maxColorAttachmentBytesPerSample = -1;
+
+EM_ASYNC_JS(int, get_limit, (), {
+  if (!navigator.gpu) {
+    throw Error("WebGPU not supported.");
+  }
+
+  const adapter = await navigator.gpu.requestAdapter();
```

Comment:
(As mentioned this adapter object could technically be a different hardware adapter than the other one which could cause the test to fail. Using GetLimits would avoid this. This is incredibly unlikely to actually cause problems though.)